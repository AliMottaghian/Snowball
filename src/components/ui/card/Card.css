.baseCard{
 background-color: var(--white);
 margin: 2em auto;   
 max-width: 600px;
 padding: 20px;
 border-radius: var(--radius);
}