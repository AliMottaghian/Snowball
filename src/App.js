import { useState, useEffect } from "react";
import Card from "./components/ui/card/Card";
import SelectTime from "./components/selectTime/SelectTime";
import AnalogClock from "./components/calculateResult/CalculateResult";
import { angleCalculator } from "../src/helpers/angleCalculator";

const App = () => {
  const [timeVal, setTimeVal] = useState(null);
  const [angleVal, setAngleVal] = useState(null);

  const pickerChange = (val) => {
    setTimeVal(val);
  };
  useEffect(() => {
    if (timeVal) {
      const result = angleCalculator(timeVal);
      setAngleVal(result);
    }
  }, [timeVal]);
  return (
    <div className="App">
      <section>
        <Card>
          <SelectTime pickerChange={pickerChange} timeVal={timeVal} />
        </Card>
      </section>
      {angleVal && timeVal && (
        <section>
          <Card>
            <AnalogClock timeVal={timeVal} angleVal={angleVal} />
          </Card>
        </section>
      )}
    </div>
  );
};

export default App;
