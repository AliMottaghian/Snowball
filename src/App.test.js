import App from "./App";
import Card from "./components/ui/card/Card";

describe("<App />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });
  it("should render App component without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should render one <Card /> when init", () => {
    const cardBox = wrapper.find(Card);
    expect(cardBox).toHaveLength(1);
  });
});
