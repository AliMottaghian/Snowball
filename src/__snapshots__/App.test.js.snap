// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render App component without crashing 1`] = `
<div
  className="App"
>
  <section>
    <Card>
      <SelectTime
        pickerChange={[Function]}
        timeVal={null}
      />
    </Card>
  </section>
</div>
`;
